scrypto-test/src/environment/env.rs

```diff
@@ -719,31 +641,47 @@ impl TestEnvironment {
     ///
     /// This method panics if the component state can not be decoded as the generic type parameter
     /// [`S`].
-    pub fn read_component_state<S, N>(&mut self, node_id: N) -> Result<S, RuntimeError>
+    pub fn with_component_state<S, N, F, O>(
```

Comment:
Looks like we need to update the rustdoc a bit.

Reply:
Yup, updated.